bundles/org.openhab.binding.bluelink/src/main/java/org/openhab/binding/bluelink/internal/handler/BluelinkAccountHandler.java

```diff
@@ -0,0 +1,160 @@
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BluelinkAccountHandler} is responsible for handling the Bluelink account bridge.
+ * It manages authentication and provides API access to vehicle handlers.
+ *
+ * @author Marcus Better - Initial contribution
+ */
+@NonNullByDefault
+public class BluelinkAccountHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(BluelinkAccountHandler.class);
+
+    private final HttpClient httpClient;
+    private final TimeZoneProvider timeZoneProvider;
+    private final ConcurrentMap<String, VehicleInfo> vehicles = new ConcurrentHashMap<>(1);
+
+    private volatile @Nullable BluelinkApi api;
+
+    public BluelinkAccountHandler(final Bridge bridge, final HttpClient httpClient,
+            final TimeZoneProvider timeZoneProvider) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Bluelink account handler");
+
+        final BluelinkAccountConfiguration config = getConfigAs(BluelinkAccountConfiguration.class);
+        final String username = config.username;
+        final String password = config.password;
+        if (username == null || username.isBlank() || password == null || password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Username and password are required");
+            return;
+        }
+
+        final String configuredBaseUrl = config.apiBaseUrl;
+        final String baseUrl = configuredBaseUrl != null ? configuredBaseUrl : API_ENDPOINT;
+        this.api = new BluelinkApi(httpClient, baseUrl, timeZoneProvider, username, password, config.pin);
+        scheduler.execute(this::login);
+    }
+
+    private void login() {
+        final BluelinkApi bluelinkApi = api;
+        if (bluelinkApi == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "API not initialized");
+            return;
+        }
+
+        try {
+            if (bluelinkApi.login()) {
+                logger.debug("Bluelink login successful");
+                loadEnrolledVehicles();
+                logger.debug("Found {} vehicles", vehicles.size());
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Login failed");
+            }
+        } catch (final BluelinkApiException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Bluelink account handler");
+        api = null;
```

Comment:
You might need to clear `scheduler` as it runs an async task. and can try to update the thing state after disposal.

Reply:
The scheduler is now cleared, but only _after_ the api is set to null. This can cause a NPE.
The login task has to be cancelled first, afterwards the api can be set to null.